src/dbnode/storage/shard.go

```diff
@@ -2849,9 +2816,99 @@ READER:
 
 	s.logger.Debug("finished aggregating tiles",
 		zap.Uint32("shard", s.ID()),
-		zap.Int64("processedTiles", processedTileCount.Load()))
+		zap.Int64("processedTiles", processedTileCount))
+
+	return processedTileCount, nil
+}
+
+func encodeAggregatedSeries(
+	seriesIter tile.SeriesFrameIterator,
+	annotationPayload annotation.Payload,
+	downsampledValues []downsampled.Value,
+	encoder encoding.Encoder,
+) (int64, error) {
+	var (
+		prevFrameLastValue = math.NaN()
+		processedTileCount int64
+		handleValueResets  bool
+		firstUnit          xtime.Unit
+		firstAnnotation    ts.Annotation
+		err                error
+	)
+
+	for seriesIter.Next() {
+		frame := seriesIter.Current()
 
-	return processedTileCount.Load(), nil
+		frameValues := frame.Values()
+		if len(frameValues) == 0 {
+			continue
+		}
+
+		if processedTileCount == 0 {
```

Comment:
nit: if we're only using this to track the start of a series, maybe a `bool` would work better?

Reply:
Not only that; also used as a return value for the whole stack of `AggregateTiles` methods (logging, etc).

Reply:
Ah oops missed that it was being returned